test/e2e-tests/support/client_set.go

```diff
@@ -0,0 +1,33 @@
+package support
+
+import (
+	"flag"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+	"path/filepath"
+)
+
+func ClusterClientSet() (*kubernetes.Clientset, error) {
+	var kubeConfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeConfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeConfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
```

Comment:
Could we just use `KUBECONFIG` environment variable here, and if needed, set the default in the Makefile?

Reply:
Is the `KUBECONFIG`enviroment variable set automatically for you with `kubectl` or `oc` atleast for me it isnt set when I log in. And setting a default in the Makefile doesn't seem right to me, I would rather all config be handled through arguments rather than enviroment variables set by a Makefile

Reply:
> Is the `KUBECONFIG` enviroment variable set automatically for you with `kubectl` or `oc` atleast for me it isnt set when I log in.

No, it's an alternative where you can point to a different kubeconfig file for different clusters/contexts, rather than needing to interactively login/switch between clusters/contexts that are specified in a single file in the default location (`~/.kube/config`).

It might be more common of a pattern in the vanilla k8s world than OpenShift, since I don't think there's even a `login` subcommand on the `kubectl` CLI (at least not on the version that I've got, which might be a little old).

I'm +1 for using `KUBECONFIG` since it's a known good approach, and potentially friendlier for running in environments like containers or CI.

Reply:
Okay I have updated the PR to do as follows:
- check if the `KUBECONFIG` env variable is set and if so it uses that location
- Else it just uses the default location `~/.kube/config`